sdks/java/harness/src/main/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClient.java

```diff
@@ -122,7 +124,9 @@ public BeamFnLoggingClient(
     logRecordHandler.setLevel(Level.ALL);
     logRecordHandler.setFormatter(DEFAULT_FORMATTER);
     logRecordHandler.executeOn(options.as(ExecutorOptions.class).getScheduledExecutorService());
```

Comment:
we have a race here, `executeOn` will start the logging thread before `outboundObserver` has been initialized.